README.md

```diff
@@ -91,16 +219,627 @@ impl DeserializeFromValue<MyError> for Name {
+    type Err = Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "*" => Wildcard::Wildcard,
+            s => Wildcard::Value(s.to_string()),
+        })
+    }
+}
+
+let data = deserialize::<Wildcard, _, JsonError>(
+    json!("doggo"),
+)
+.unwrap();
+assert_eq!(data, Wildcard::Value(String::from("doggo")));
+
+let data = deserialize::<Wildcard, _, JsonError>(
+    json!("*"),
+)
+.unwrap();
+assert_eq!(data, Wildcard::Wildcard);
+```
+
+##### Or as a field attribute
+
+```rust
+use deserr::{DeserializeFromValue, deserialize};
+use deserr::serde_json::JsonError;
+use serde_json::json;
+use std::convert::Infallible;
+use std::str::FromStr;
+use std::num::ParseIntError;
+
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+struct Search {
+    query: String,
+    #[deserr(from(&String) = FromStr::from_str -> ParseIntError)]
+    limit: usize,
+
+}
+
+let data = deserialize::<Search, _, JsonError>(
+    json!({ "query": "doggo", "limit": "12" }),
+)
+.unwrap();
+assert_eq!(data, Search { query: String::from("doggo"), limit: 12 });
+
+let error = deserialize::<Search, _, JsonError>(
+    json!({ "query": "doggo", "limit": 12 }),
+)
+.unwrap_err();
+assert_eq!(error.to_string(), "invalid type: Integer `12`, expected a String at `.limit`.");
+```
+
+#### `validate`
+
+Validate a structure **after** it has been deserialized.
+This is typically useful when your validation logic needs to take multiple fields into account.
+
+```rust
+use deserr::{DeserializeFromValue, DeserializeError, ErrorKind, ValuePointerRef, deserialize};
+use deserr::serde_json::JsonError;
+use serde_json::json;
+use std::convert::Infallible;
+
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+#[deserr(validate = validate_range -> __Deserr_E)]
+struct Range {
+    min: u8,
+    max: u8,
+}
+
+fn validate_range<E: DeserializeError>(
+    range: Range,
+    location: ValuePointerRef,
+) -> Result<Range, E> {
+    if range.min > range.max {
+        Err(deserr::take_result_content(E::error::<Infallible>(
+            None,
+            ErrorKind::Unexpected {
+                msg: format!(
+                    "`max` (`{}`) should be greater than `min` (`{}`)",
+                    range.max, range.min
+                ),
+            },
+            location,
+        )))
+    } else {
+        Ok(range)
+    }
+}
+
+let data = deserialize::<Range, _, JsonError>(
+    json!({ "min": 2, "max": 4 }),
+)
+.unwrap();
+assert_eq!(data, Range { min: 2, max: 4 });
+
+let error = deserialize::<Range, _, JsonError>(
+    json!({ "min": 4, "max": 2 }),
+)
+.unwrap_err();
+assert_eq!(error.to_string(), "`max` (`2`) should be greater than `min` (`4`) at ``.");
+```
+
+#### `default`
+
+Allows you to specify a default value for a field.
+
+Note that, by default, `Option` doesn't automatically use this attribute, as it is the case with serde.
+Here you need to explicitly define whether your type can get a default value.
```

Comment:
I kind of forgot the exact rationale. But I think there were two reasons:
1. Simplify the `DeserializeFromValue` trait, so that it doesn't have to tell what a type should deserialise to when the field is missing from the payload.
2. Make it so that optional fields can also be mandatory easily.

But also I think we found it better to always be explicit about it, so it is also a question of preference.

Reply:
I think it's OK to have this coming down to preference, but I also think parts of the rationale (reasons (1) and (2)) could be provided to users :-)